reference/features/security.md

```diff
@@ -0,0 +1,272 @@
+
+## API keys
+
+API keys are necessary to access the majority of API routes in a protected MeiliSearch instance. When creating API keys, you can set exactly which indexes, endpoints, and routes they can access, as well how long the key should remain valid.
+
+### Creating an API key
+
+You can create API keys by using the [`POST /keys` endpoint](/reference/api/keys.md#create-a-key). This endpoint is always protected and can only be accessed with the `master` key.
+
+This endpoint accepts four arguments:
+
+- `indexes`: an array of strings indicating the `uid`s of indexes the key can access _(mandatory)_
+- `actions`: an array of [valid key actions](/reference/api/keys.md#actions) _(mandatory)_
+- `expiresAt`: a valid ISO 8601 datetime after which the key will stop working _(mandatory)_
+- `description`: a string describing the key's intended use _(optional)_
+
+`indexes`, `actions`, and `expiresAt` are mandatory. Failing to specify any of these arguments when creating a key will result in an error.
+
+::: note
+While we strongly advise against it, you can create an API key with no expiry date by explicitly passing a `null` value to `expiresAt`.
+:::
+
+#### Example
+
+You can create an API key that can only perform searches on your `scifi_books` index. Since you expect this key to remain useful for a long time and it doesn't have write access to any other endpoints, you decide you set its expiry date to the next year:
+
+<CodeSamples id="security_guide_create_key_1" />
+
+All [`/keys` endpoints](/reference/api/keys.md) are synchronous, so your key will be generated immediately:
+
+```json
+{
+  "description": "Search key",
+  "key": "d0552b41536279a0ad88bd595327b96f01176a60c2243e906c52ac02375f9bc4",
+  "actions": [
+      "search"
+  ],
+  "indexes": [
+  "scifi_books"
+  ],
+  "expiresAt": "2023-01-01T00:00:00Z",
+  "createdAt": "2022-01-01T10:00:00Z",
+  "updatedAt": "2022-01-01T10:00:00Z"
+}
+```
+
+Before you start searching, you need to add content to that index, so you create another key that allows you to add, update, and delete documents. Since this is a sensitive task, you set the key to expire the next day:
+
+<CodeSamples id="security_guide_create_key_2" />
+
+Once again, you'll receive the key in the request response body:
+
+```json
+{
+  "description": "Manage sci-fi books key",
+  "key": "d0552b41536279a0ad88bd595327b96f01176a60c2243e906c52ac02375f9bc4",
+  "actions": [
+    "documents.add",
+    "documents.get",
+    "documents.delete"
+  ],
+  "indexes": [
+  "scifi_books"
+  ],
+  "expiresAt": "2022-02-01T00:00:00Z",
+  "createdAt": "2022-01-01T10:00:00Z",
+  "updatedAt": "2022-01-01T10:00:00Z"
+}
+```
+
+### Default API Keys
+
+When you launch an instance with a `master` key for the first time, MeiliSearch will automatically generate two API keys: `Default Search API Key` and `Default Admin API Key`.
```

Comment:
@gmourier on another comment:

> This is to speed up the process for a user who wants to secure his instance without creating special keys. We generate two keys identical to the old public and private keys. We just make the system more consistent by transferring them as API Key resources while the user can keep it fast (thus not having to create these default API Keys) for a simple security use-case.

Not sure I completely follow. What is the point of creating the keys before a user has signaled they will need them? And won't those keys change the moment the user finally sets a master key, since AFAIK changing a master key regenerates all existing keys?